### PR TITLE
Implement Lanczos image filter

### PR DESCRIPTION
**Pull request by Daw11**
This pull adds the Lanczos filter to the Image class.

It's one of the most common filters and is present in almost every software that manipulates images.

It's a lot slower and require more resources than all the other, but it generates the best downscaled images.

This pull makes it the default filter used by the scene previews( the ones that appear when you hover with the mouse on the scene tabs ).

This is a before and after of the same scene preview:
before-after

It can also be used as a normal filter in GDScript with:
`image.resize( 200, 200, Image.INTERPOLATE_LANCZOS )`

### CODE REVIEW

**Comment by Daw11** [comment]
I'm updating a bit this function to make it work more like the implementation described in wikipedia: https://en.wikipedia.org/wiki/Lanczos_resampling

**Comment by Daw11** [comment]
I have completed all the changes.

**Comment by Daw11** [comment]
The two continue inside the for loops can probably be removed, I'll push a change later.

**Comment by Daw11** [comment]
I completed all the changes.

File: core/image.cpp [hunk 1]
```diff
@@ -725,6 +725,129 @@ static void _scale_nearest(const uint8_t *__restrict p_src, uint8_t *__restrict
...
+
+	int32_t src_width = p_src_width;
+	int32_t src_height = p_src_height;
+	int32_t dst_height = p_dst_height;
+	int32_t dst_width = p_dst_width;
+
+	float y_scale = float(p_src_height) / float(p_dst_height);
+	float x_scale = float(p_src_width) / float(p_dst_width);
+
+	float *buffer = new float[src_height * dst_width * CC]; // Store the first pass in a buffer
```
Comment (ShyRed) [inline core/image.cpp]: You might want to replace the usages of `new` and `delete` with Godot's own functions.  See https://docs.godotengine.org/en/3.1/development/cpp/core_types.html#allocating-memory for info.

Comment (Daw11) [reply core/image.cpp]: Thank you, I have updated the code.

**Review by reduz — APPROVED** [review]

**Review by mhilbrunner — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Comment by mhilbrunner** [comment]
Ah - yup, thanks! Here's to less blurry image previews in the editor in the feature! :)

**Pull request merged**